### PR TITLE
Added type definitions for react-draft-wysiwyg

### PR DESCRIPTION
**Pull request by imechZhangLY**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/react-draft-wysiwyg/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+// Definitions by: imechZhangLY <https://github.com/imechZhangLY>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import * as React from 'react';
+import * as Draft from 'draft-js';
+
+export as namespace Wysiwyg;
+
+declare namespace Wysiwyg {
```
Comment (andy-ms) [inline types/react-draft-wysiwyg/index.d.ts]: The namespace is unnecessary. You can just export everything directly as in `export type SyntheticKeyboardEvent = React.KeyboardEvent<{}>; export type SyntheticEvent = ...;`.

File: types/react-draft-wysiwyg/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        customBlockRenderFunc?(block: ContentBlock): any;
+        wrapperId?: number;
+        customDecorators?: object[];
+        editorRef?(ref: object): void;
+        handlePastedText?(text: string, html: string, editorState: EditorState, onChange: (editorState: EditorState) => void): boolean;
+    }
+
+    class Editor extends React.Component<EditorProps, {}> {
+        focusEditor(): void;
+    }
```
Comment (andy-ms) [inline types/react-draft-wysiwyg/index.d.ts]: Everything past this line is unnecessary if the namespace is removed.

File: types/react-draft-wysiwyg/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": false,
```
Comment (andy-ms) [inline types/react-draft-wysiwyg/tsconfig.json]: This should be `true` for all new packages.

File: types/react-draft-wysiwyg/tslint.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (andy-ms) [inline types/react-draft-wysiwyg/tslint.json]: For a new package this file should just be `{ "extends": "dtslint/dt.json" }` with no disables.

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/react-draft-wysiwyg/tsconfig.json (hunk 3 above)
Comment (sheetalkamat) [reply types/react-draft-wysiwyg/tsconfig.json]: This still isn't set to true

Comment (sheetalkamat) [reply types/react-draft-wysiwyg/tsconfig.json]: This still isn't set to true

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**